vars/runJenkinsTests.groovy

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env groovy
+
+def call(Map options = [:]) {
+    def jenkinsAdminPassword = sh(returnStdout: true, script: options.sshCommand + " sudo cat /var/lib/jenkins/secrets/initialAdminPassword").trim()
+    def version = null
+    try {
+        // NOTE: The password will be printed out in the logs, but that's fine since you still have to ssh onto the VM to use it
+        version = sh(returnStdout: true, script: 'curl --silent "' + options.utilsLocation + 'jenkins/run-cli-command.sh" | sudo bash -s -- -c "version" -ju "admin" -jp ' + jenkinsAdminPassword).trim()
+        echo "Jenkins version: " + version
+    } catch (e) {
+    }
+
+    if (!version || version == "") {
+        error("Failed to get Jenkins version.")
+    }
+
+    //verify that the Azure landing page works
+    def remoteURL = options.sshCommand.split('@')[1].split(' ')[0]
+    print remoteURL;
+
+    def compare_script_path = 'scripts/compare-web-pages.sh'
+    sh 'sudo chmod +x ' + compare_script_path
+
+    //we're using the login page, because http://localhost doest javascript redirect to login
+    diff_result = sh(returnStatus: true, script: compare_script_path + " -f http://localhost:8080/login -s http://" + remoteURL + "/login")
+    if (diff_result != 1) {
+        error("The public landing page is identical with the local page!")
+    }
+
+    return_code = sh(returnStatus: true, script: " curl --connect-timeout 10 http://" + remoteURL + ":8080")
+    if (return_code != 28) {
+        error("Port 8080 is not closed!")
```

Comment:
Suggest "Port 8080 is accessible over the public IP address, possibly exposing the user to Man in the Middle Attacks!"